src/domain/Session.ts

```diff
@@ -15,6 +15,21 @@ const { jws } = KJUR;
 const swarmKey = <unknown>(KEYUTIL.getKey(swarmPublicKey)) as string;
 const MINIMUM_WAZO_ENGINE_VERSION_FOR_DEFAULT_CONTEXT = '19.08';
 
+type SessionMetadata = {
+  jwt?: string;
+  username?: string;
+  uuid: UUID;
+  tenant_uuid?: UUID;
+  xivo_user_uuid?: UUID;
+  groups?: Array<string>;
+  xivo_uuid?: UUID;
+  tenants?: Array<{
+    uuid: UUID;
+  }>;
+  auth_id?: UUID;
+  [key: string]: any; // Plugin can extend metadata
```

Comment:
Would it be better to put the plugin extendable `[key: string]` in a separate `extraMetadata` param or just an `extra` key in that same type ? (type safety)